qiita_pet/handlers/study_handlers/listing_handlers.py

```diff
@@ -57,59 +126,27 @@ def _build_study_info(user, results=None):
-    for row, info in enumerate(study_info):
+    for info in study_info:
+        # Convert DictCursor to proper dict
+        info = dict(info)
         study = Study(info['study_id'])
-        status = study.status
-        # Just passing the email address as the name here, since
-        # name is not a required field in qiita.qiita_user
-        PI = StudyPerson(info['principal_investigator_id'])
-        PI = study_person_linkifier((PI.email, PI.name))
-        if info['pmid'] is not None:
-            pmids = ", ".join([pubmed_linkifier([p])
-                               for p in info['pmid']])
-        else:
-            pmids = ""
-        if info["number_samples_collected"] is None:
-            info["number_samples_collected"] = "0"
-        shared = _get_shared_links_for_study(study)
-        meta_complete_glyph = "ok" if info["metadata_complete"] else "remove"
-        # build the HTML elements needed for table cell
-        title = ("<a href='#' data-toggle='modal' "
-                 "data-target='#study-abstract-modal' "
-                 "onclick='fillAbstract(\"studies-table\", {0})'>"
-                 "<span class='glyphicon glyphicon-file' "
-                 "aria-hidden='true'></span></a> | "
-                 "<a href='/study/description/{1}' "
-                 "id='study{0}-title'>{2}</a>").format(
-                     str(row), str(study.id), info["study_title"])
-        meta_complete = "<span class='glyphicon glyphicon-%s'></span>" % \
-            meta_complete_glyph
-        if status == 'public':
-            shared = "Not Available"
-        else:
-            shared = ("<span id='shared_html_{0}'>{1}</span><br/>"
-                      "<a class='btn btn-primary btn-xs' data-toggle='modal' "
-                      "data-target='#share-study-modal-view' "
-                      "onclick='modify_sharing({0});'>Modify</a>".format(
-                          study.id, shared))
-
-        infolist.append({
-            "checkbox": "<input type='checkbox' value='%d' />" % study.id,
-            "id": study.id,
-            "title": title,
-            "meta_complete": meta_complete,
-            "num_samples": info["number_samples_collected"],
-            "shared": shared,
-            "num_raw_data": len(study.raw_data()),
-            "pi": PI,
-            "pmid": pmids,
-            "status": status,
-            "abstract": info["study_abstract"]
-
-        })
+        # Build the processed data info for the study if none passed
+        if build_samples:
+                proc_data = study.processed_data()
+                proc_samples = {}
+                study_proc = {study.id: proc_data}
+                for pid in proc_data:
+                    proc_samples[pid] = ProcessedData(pid).samples
+
+        study_info = _build_single_study_info(study, info)
+        # Build the proc data info list for the child row in datatable
+        study_info["proc_data_info"] = _build_proc_data_info(study, study_proc,
```

Comment:
The problem is that there is no separation. The "proc_data_info" is added
to the "study_info" dictionary, which means that the proc_data_info is part
of the study_info. That is why I think it has to be added to the
"_build_sinfle_study_info", because that is info that belongs to the study.
You're just building it ad-hoc inside the other function, instead of
building it in 2 different places. My point is that if at some point we
need the "_build_single_study_info" in another part of the code, we will
need to remember to call "_build_proc_data_info" independently, which shows
a deficiency on the code structure design.

2015-04-12 16:48 GMT-07:00 Joshua Shorenstein <notifications@github.com>:

> In qiita_pet/handlers/study_handlers/listing_handlers.py
> <https://github.com/biocore/qiita/pull/1039#discussion_r28208973>:
>
> > -            "pmid": pmids,
> > -            "status": status,
> > -            "abstract": info["study_abstract"]
> > -
> > -        })
> > +        # Build the processed data info for the study if none passed
> > +        if build_samples:
> > +                proc_data = study.processed_data()
> > +                proc_samples = {}
> > +                study_proc = {study.id: proc_data}
> > +                for pid in proc_data:
> > +                    proc_samples[pid] = ProcessedData(pid).samples
> > +
> > +        study_info = _build_single_study_info(study, info)
> > +        # Build the proc data info list for the child row in datatable
> > +        study_info["proc_data_info"] = _build_proc_data_info(study, study_proc,
>
> This is to make a clear separation between proc data and study info. It
> also makes it so I don't have to pass as many things into the
> single_study_info function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1039/files#r28208973>.
>



-- 
Jose Navas